unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/KernelCommunication/RPC/Services/CRDTService/CRDTService.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Generic;
+using Cysharp.Threading.Tasks;
+using DCL;
+using DCL.CRDT;
+using Google.Protobuf;
+using KernelCommunication;
+using rpc_csharp;
+using UnityEngine;
+
+namespace RPC.Services
+{
+    public static class CRDTServiceImpl
+    {
+        private static readonly UniTask<CRDTResponse> defaultResponse = UniTask.FromResult(new CRDTResponse());
+        private static readonly CRDTManyMessages reusableCrdtMessage = new CRDTManyMessages();
+
+        private static readonly CRDTStream crdtStream = new CRDTStream();
+
+        public static void RegisterService(RpcServerPort<RPCContext> port)
+        {
+            CRDTService<RPCContext>.RegisterService(
+                port,
+                sendCrdt: OnCRDTReceived,
+                crdtNotificationStream: CRDTNotificationStream
+            );
+        }
+
+        private static UniTask<CRDTResponse> OnCRDTReceived(CRDTManyMessages messages, RPCContext context)
```

Comment:
We need a cancellation token here, take a look at other usages (Avatar.cs for example) to implement it properly using `OperationCanceledException`, otherwise a cancelled task will keep runing until the end of player loop.

Reply:
No sure if I understand.
In `Avatar.cs` we are cancelling the task if fails to load or if disposed while loading.
Here we are not cancelling the tasks since it's just deserializing and adding to a queue, so we don't have any of those kind of conditions where we should cancel the running task